### PR TITLE
Basic Undead Trait

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Added **_Undead_** trait. They can't have children. They're hated by religions with the Death Magic Shunned or Criminal. They change their skin color to green rotting.
- The Mashani (a tauren religion) has a special doctrine that makes them never hate the undead.

## Notes:
Example of a random human before and after adding the `being_undead` trait to them.

<details>
<summary>Click to expand</summary>

image
image

</details>

Characters with the trait aren't all green. Their skin color is chosen randomly from a range:

<details>
<summary>Click to expand</summary>

image
image

</details>

## How to test:
- `add_trait being_undead` adds the trait to your character.

Test the trait in-game, its opinions, and its looks.

### CODE REVIEW

**Comment by zumbak04** [comment]
Overall, looks good, but I would like to implement #161 first.

File: common/traits/wc_undead_trait.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿being_undead = {
```
Comment (zumbak04) [inline common/traits/wc_undead_trait.txt:1]: I don't see the icon.
<details>
<summary>Click to expand</summary>

image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/traits/wc_undead_trait.txt (hunk 1 above)
Comment (ercarp) [reply common/traits/wc_undead_trait.txt:1]: > 
> 
> I don't see the icon.

Fixed in 69bdd7c46661cb7a8f0e151fcaaa0a659fe85086 and tested that it's working even after switching branches.

<details>
<summary>Click to expand</summary>

image


</details>

**Review by Deuce20201 — COMMENTED** [review]
Question: Is there no undead religion yet? When I made Genn undead his religion stayed the same.

**Comment by zumbak04** [comment]
> Question: Is there no undead religion yet? When I made Genn undead his religion stayed the same.

You don't change the religion on becoming undead. Otherwise, we would mention it in the changelog.

**Review by Deuce20201 — APPROVED** [review]
In that case everything looks fine. I tested it on every custom race added so far and they all look good and color varies. And evreyone hates me when I am undead except for those Taurens you mentioned.

**Review by zumbak04 — APPROVED** [review]
LGTM!

**Review by TheDrPibb1999 — APPROVED** [review]
Look great and worked on every race, the only other people I found who did not hate my undead was the Drakkari in Northrend, and well as people have previously said the Tauren. Nice trait and icon :)

**Pull request merged**